src/super_gradients/examples/qat_from_recipe_example/qat_from_recipe.py

```diff
@@ -0,0 +1,220 @@
+    fuse_repvgg_blocks_residual_branches(model)
+    model.to(device=device)  # make sure that reparametrized modules are on the same device
+
+    q_util.quantize_module(model)
+    model.to(device=device)  # we have new modules, they can be on different devices
+
+    calibration_method = calibration_method or method_i
+    calibrator.calibrate_model(
+        model,
+        method=calibration_method,
+        calib_data_loader=calibration_dataloader,
+        num_calib_batches=num_calib_batches,
+        percentile=percentile,
+    )
+    model.to(device=device)  # we can have _amax buffers scattered over different devices
+
+    model.train()
+    return model
+
+
+def modify_training_params_for_qat(cfg):
+    # Q/DQ Layers take a lot of space for activations in training mode
+    if cfg.quantization_params.selective_quantizer_params.learn_amax:
+        cfg.dataset_params.train_dataloader_params.batch_size //= 2
+        cfg.dataset_params.val_dataloader_params.batch_size //= 2
+
+    # 10% of the training regime
+    cfg.training_hyperparams.max_epochs //= 10
+
+    # very small initial LR and WD
+    lr_decay_factor = cfg.quantization_params.default_qat_recipe.lr_decay_factor
+
+    cfg.training_hyperparams.initial_lr *= lr_decay_factor
+    if cfg.training_hyperparams.warmup_initial_lr is not None:
+        cfg.training_hyperparams.warmup_initial_lr *= lr_decay_factor
+    else:
+        cfg.training_hyperparams.warmup_initial_lr = cfg.training_hyperparams.initial_lr * 0.01
+
+    cfg.training_hyperparams.optimizer_params.weight_decay *= lr_decay_factor
+
+    # as recommended by pytorch-quantization docs
+    cfg.training_hyperparams.lr_mode = "cosine"
+    cfg.training_hyperparams.lr_warmup_epochs = (cfg.training_hyperparams.max_epochs // 10) or 1
+    cfg.training_hyperparams.cosine_final_lr_ratio = 0.01
+
+    # do mess with Q/DQ
+    if cfg.training_hyperparams.ema:
+        logger.warning("EMA will be disabled for QAT run.")
+        cfg.training_hyperparams.ema = False
+
+    if cfg.training_hyperparams.sync_bn:
+        logger.warning("SyncBatchNorm will be disabled for QAT run.")
+        cfg.training_hyperparams.sync_bn = False
+
+    if len(cfg.training_hyperparams.phase_callbacks) > 0:
+        logger.warning(f"Recipe contains {len(cfg.training_hyperparams.phase_callbacks)} phase callbacks. All of them will be disabled.")
+        cfg.training_hyperparams.phase_callbacks = []
+
+    if cfg.multi_gpu != "OFF" or cfg.num_gpus != 1:
+        logger.warning(f"Recipe requests multi_gpu={cfg.multi_gpu} and num_gpus={cfg.num_gpus}. Changing to multi_gpu=OFF and num_gpus=1")
+        cfg.multi_gpu = "OFF"
+        cfg.num_gpus = 1
+
+    # no augmentations
+    cfg.dataset_params.train_dataset_params.transforms = cfg.dataset_params.val_dataset_params.transforms
+
+
+@hydra.main(config_path=pkg_resources.resource_filename("super_gradients.recipes", ""), version_base="1.2")
+def main(cfg: DictConfig) -> None:
```

Comment:
Please refactor this into 2 methods: one has Hydra's decorator, instantiation of the cfg, and then calls the other function that accepts an instantiated cfg.

```python
def qat_from_config(cfg):
   if "quantization_params" not in cfg:
        raise ValueError("Your recipe does not have quantization_params. Add them to use QAT.")
    ...

@hydra(...)
def main(cfg):
   cfg = hydra.utils.instantiate(cfg)
   return qat_from_config(cfg)
```
This way we can reuse the function that accepts a "ready" cfg.